torchsynth/synth.py

```diff
@@ -226,6 +226,16 @@ def randomize(self, seed: Optional[int] = None):
         for module in self._modules:
             self._modules[module].seed = seed
 
+    def on_post_move_to_device(self) -> None:
+        """
+        LightningModule trigger after this Synth has been moved to a different device.
+        Use this to update children SynthModules device settings
+        """
+        self.synthglobals.to(self.device)
+        for module in self.modules():
+            if isinstance(module, SynthModule):
+                module.update_device(self.device)
```

Comment:
Okay cool but why not just module.to(self.device)

Reply:
soo this is another weird one. When you call `module.to(self.device)` on the module it goes through the process of moving all the parameters to the GPU -- BUT, those parameters have already been handled by this Synth because all parameters and buffers are known to it. So we really just want to update the stuff that we are managing. It is not only redundant to call it again, but it actually has a negative affect on performance during run. I'm not entirely clear on that part.

Reply:
Really has a negative impact? I thought stuff already on the device is a no-op.
If so can you just add this comment?

Reply:
Will make a comment and an issue to check out. I'm not sure what's going on there -- might be the override on to in SynthModule?